lib/blocks.php

```diff
@@ -27,6 +27,39 @@ function gutenberg_get_registered_social_link_blocks() {
 	return $registered_social_link_blocks;
 }
 
+/**
+ * Registers blocks using block manifests discovered in block library.
+ *
+ * @since 6.9.0
+ */
+function gutenberg_register_block_types() {
+	$registry = WP_Block_Type_Registry::get_instance();
+
+	$block_manifests = glob( dirname( dirname( __FILE__ ) ) . '/packages/block-library/src/*/block.json' );
+	foreach ( $block_manifests as $block_manifest ) {
```

Comment:
Should some level of validation be happening here?

Reply:
> Should some level of validation be happening here?

Do you mean that it has necessary properties to consider it a valid block manifest?

There are some simple checks below, both to account that the file could be parsed as JSON, and that it has a `name`. We could expand on this.

https://github.com/WordPress/gutenberg/blob/1e359a8d467b23c33c664f6c277c2549866be2dd/lib/blocks.php#L41

Reply:
Validation of types, like it string, int, array etc?

Reply:
> Validation of types, like it string, int, array etc?

If I understand you correctly, we probably should want something like what exists in [`WP_Block_Type#prepare_attributes_for_render`](https://github.com/WordPress/wordpress-develop/blob/6cf77dbbb6d6b289d81dd2d33c742d150129b774/src/wp-includes/class-wp-block-type.php#L140-L157), though I expect this would be applied at the time that `$attributes` are being sourced.